linode/manager
# refactor: [M3-6852] - Remove global error interceptors

## Description 📝
Today the Cloud Manager uses a global error interceptor to add links to `APIError`s that call for users to open a support ticket. This interceptor introduces type unsafety (since it necessitates [a cast](https://github.com/linode/manager/blob/2c891f971f51e18f007ec220db003f33dae84639/packages/manager/src/utilities/interceptAPIError.tsx#L18)) and reduces flexibility in displaying error messages.

#10684 made progress towards reducing our reliance on this interceptor by using the `<ErrorMessage />` component, which formats errors at the component level. This PR completes that refactor by removing the two remaining cases where errors were being intercepted:

* Errors about the user reaching their thing limit
    * These are now handled at the component level for all entities (Linodes, firewalls, volumes, etc.)
* Errors caused when trying to migrate in an account with the `dc_migrations_disabled` customer tag
    * This is now handled in the `MigrateLinode` component

## Changes  🔄
- Deletes the global error interceptor in `request.tsx` and associated utilities (`interceptAPIError.ts`)
- Detect migration errors in the `ErrorMessage` component
- Add `ErrorMessage` in a couple of missing places (`GenerateFirewallDialog`, `MigrateLinode`)

## Target release date 🗓️
9/17

## How to test 🧪

*In all of the following instances, displayed errors should include a link to the `SupportTicketDialog`. For example:*

![image](https://github.com/user-attachments/assets/92ce6591-c9a5-473a-89a8-a20fe3bde093)


* Add the `dc_migrations_disabled` tag to your account and attempt to migrate a Linode
* Reduce your thing limit/reputation in admin and attempt to create new services (e.g., Linodes, volumes, firewalls, etc.)
* Verify `APIError`s otherwise appear as expected across the app
